### PR TITLE
Restore plan

### PR DESCRIPTION
**Pull request by LeoCBS**

### CODE REVIEW

File: backup/restore/restore.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	session, err := mgo.Dial(mongoURL)
+	if err != nil {
+		return err
+	}
+	defer session.Close()
+
+	session.SetSafe(&mgo.Safe{})
+	c := session.DB(database).C(collection)
+
+	countRestored, err2 := c.Find(bson.M{}).Count()
```
Comment (kamilash) [inline backup/restore/restore.go]: I think only c.Count() also works.

File: backup/restore/restore.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	if err2 != nil {
+		return err
```
Comment (kamilash) [inline backup/restore/restore.go]: why not return err2?

File: backup/backup.go [hunk 3]
```diff
@@ -28,6 +29,13 @@ func Run(plan config.Plan, tmpPath string, storagePath string) (Result, error) {
 		return res, err
 	}
 
+	restoreOutput, err := restore.Restore(plan, archive)
+	if err != nil {
+		return res, err
+	} else {
```
Comment (zbioe) [inline backup/backup.go]: maybe this `else` is not necessary

File: backup/restore/restore.go [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+	if err != nil {
+		return "", err
+	}
+	defer session.Close()
+	defer cleanMongo(session)
+
+	session.SetSafe(&mgo.Safe{})
+	for _, collec := range r.Collections {
+		c := session.DB(database).C(collec.Name)
+		countRestored, err2 := c.Find(bson.M{}).Count()
```
Comment (zbioe) [inline backup/restore/restore.go]: maybe use `var countRestored int`  for reuse the var err and don't need err2

File: backup/backup.go (hunk 3 above)
Comment (LeoCBS) [reply backup/backup.go]: indeed :) thanks

File: backup/restore/restore.go (hunk 2 above)
Comment (LeoCBS) [reply backup/restore/restore.go]: my mistake.. correctly is return err2 :)

File: backup/restore/restore.go (hunk 4 above)
Comment (LeoCBS) [reply backup/restore/restore.go]: yeap :)

File: backup/restore/restore.go (hunk 1 above)
Comment (LeoCBS) [reply backup/restore/restore.go]: nice.. and we could remove bson import :)

Comment (kamilash) [reply backup/restore/restore.go]: even better then :D

**Review by cadicallegari — APPROVED** [review]

**Pull request merged**